pkg/logutil/logutil.go

```diff
@@ -30,3 +31,23 @@ func NewWriterToFile(maxSize int, dirPath, name string) (io.Writer, error) {
 	return lw, nil
 
 }
+
+// TeeOutput tees out output to a file path specified by env var `UROOT_LOG_PATH` up to a max limit. Can optionally create the directory structure for the filepath.
+func TeeOutput(writer io.Writer, maxSize int, initDir bool) (io.Writer, error) {
+	logPath := os.Getenv("UROOT_LOG_PATH")
+	if logPath != "" {
+		if initDir {
```

Comment:
if `logPath` is not empty, and `initDir` is false, then further code is executed with `logPath` directory not present, right ? -- Do you anticipate such case will be needed ? 

Then the best course of action for user is make sure UROOT_LOG_PATH is specified in kernel cmdline, and pass in `initDir` as true. If this is always the case we want, might be worth simply always create a directory when not present (and more user friendly way, check if it exist, if not, log a notice, and create one). Given this will be the top level setup function, which is above `NewWriterToFile` in call stack, so I think create directory is fine here

Reply:
sgtm! I think it's reasonable that the user verifies UROOT_LOG_PATH points to the right location so that it doesn't create unintended directories.

Reply:
precisely